rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -550,10 +554,14 @@ private int retryGetEndStatus(URI endPath, boolean compensate) {
                      */
                     status = compensate ? ParticipantStatus.Compensated : ParticipantStatus.Completed;
                     return TwoPhaseOutcome.FINISH_OK;
-                } else if (response.getStatus() == Response.Status.ACCEPTED.getStatusCode() ||
-                        Response.Status.Family.familyOf(response.getStatus()).equals(
-                            Response.Status.Family.SERVER_ERROR)) {
-                    // these response codes indicate that the implementation should retry later
+                } else if (response.getStatus() == Response.Status.ACCEPTED.getStatusCode()) {
+                    // this response code indicate that the implementation should retry later
+                    lra.setLRAStatusReason(Transaction.LRAStatusReason.PARTICIPANT_RESPONSE);
+                    return TwoPhaseOutcome.HEURISTIC_HAZARD;
+                } else if (Response.Status.Family.familyOf(response.getStatus())
+                    .equals(Response.Status.Family.SERVER_ERROR)) {
+                    // participant processing failed, retry later
+                    lra.setLRAStatusReason(Transaction.LRAStatusReason.PARTICIPANT_FAILED_STATUS);
```

Comment:
Won't this status reason overwrite the existing reason (the same comment applies to the other places where the participant record updates the LRA status reason).